cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -2805,6 +2815,22 @@
             (_ Unit (emit (MInst.XmmRmR (SseOpcode.Maxpd) x y dst))))
         dst))
 
+;; Helper for creating `vfmadd213ps` instructions.
+(decl x64_vfmadd213ps (Xmm Xmm XmmMem) Xmm)
+(rule (x64_vfmadd213ps x y z)
+      (let ((dst WritableXmm (temp_writable_xmm))
+            (_1 Unit (emit (MInst.XmmUnaryRmR (SseOpcode.Movups) x dst)))
+            (_2 Unit (emit (MInst.XmmRmRVex (AvxOpcode.Vfmadd213ps) z y dst))))
+        dst))
+
+;; Helper for creating `vfmadd213pd` instructions.
+(decl x64_vfmadd213pd (Xmm Xmm XmmMem) Xmm)
+(rule (x64_vfmadd213pd x y z)
+      (let ((dst WritableXmm (temp_writable_xmm))
+            (_1 Unit (emit (MInst.XmmUnaryRmR (SseOpcode.Movups) x dst)))
```

Comment:
How come we need these moves?

Reply:
Ah, I suspect this may be due to using a "mod" operand rather than a separate src and dest with dest tied to src via a reuse-constraint. @afonso360 would you mind matching how the other instructions work in that way? Then you can assert that src1 == dst during emission and the emission code remains unchanged, but it makes the usage here simpler.

Reply:
They aren't actually emitted, but `x` isn't writable, so we need a reg that is, and since `vfmadd213pd` modifies the first reg we also need the value of `x` to be in that register. I'm not sure if this is the best way to achieve that.